opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -392,3 +404,25 @@ impl MetricReader for PeriodicReader {
         shutdown_result
     }
 }
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use super::PeriodicReader;
+    use crate::{
+        metrics::data::ResourceMetrics, metrics::reader::MetricReader, runtime,
+        testing::metrics::InMemoryMetricsExporter, Resource,
+    };
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn unregistered_collect() {
+        let exporter = InMemoryMetricsExporter::default();
+        let reader = PeriodicReader::builder(exporter.clone(), runtime::Tokio).build();
+
+        let mut rm = ResourceMetrics {
+            resource: Resource::empty(),
+            scope_metrics: Vec::new(),
+        };
+        let result = reader.collect(&mut rm);
+        result.expect_err("error expected when reader is not registered");
+    }
+}
```

Comment:
Do you think it would be easy to add the test for success scenario (something like):
 - create PeriodicReader 
 - Create SdkMeterProvider instance / or Pipelines with the above created reader.
 - This should invoke worker, and collect cycle.
 
If not straightforward, we can park it for later. (Ideally, this test should have existed at first place before this PR.)

Reply:
I'll give it a shot :+1: